data/spells/lib/spells.lua

```diff
@@ -296,3 +306,74 @@ function Creature:addDamageCondition(target, conditionType, listType, damage, ti
 	target:addCondition(condition)
 	return true
 end
+
+function isOnCombatArea(casterPosition, targetPosition, coordinates)
+	if casterPosition.z == targetPosition.z then
+		for _, axis in ipairs(coordinates) do
+			if casterPosition.x + axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x + axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			end
+		end
+	end
+	return false
+end
+
+function Creature:addPartyCondition(combat, variant, condition, baseMana)
+	local position = self:getPosition()
+	local party = self:getParty()
+	if not party then
+		self:sendTextMessage(MESSAGE_STATUS_SMALL, "No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local membersList = party:getMembers()
+	membersList[#membersList + 1] = party:getLeader()
```

Comment:
It's been like this since tfs 1.0, although ``table.insert`` states better what it does, and in this case I think it would be cool to use it (performance related) but probably not in line 348??